src/inmanta/db/versions/v202409090.py

```diff
@@ -0,0 +1,29 @@
+"""
+    Copyright 2024 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+from asyncpg import Connection
+
+
+async def update(connection: Connection) -> None:
+    """
+    Add indexes required by the diagnose endpoint.
+    """
+    schema = """
+    CREATE INDEX report_started_compile_returncode ON report(started DESC, compile, returncode);
```

Comment:
Are you sure that this index is actually used by the query? I'm surprised to see it here, since we first select on compile and returncode, and only then sort on `started`. Futhermore, I'd expect the set of reports to sort on to be either one or none.

Reply:
You are right. I swaped it around. It should have been `(compile, returncode, started DESC)`. I'll change that. Thanks!

Reply:
But even then, is the index worth it? Do we ever expect the `compile, returncode` match to return more than one record?

Reply:
i.e. we stop after the first failed compile stage, don't we?